cluster/manifests/z-karpenter/01-rbac.yaml

```diff
@@ -129,7 +129,7 @@ rules:
   # Read
   - apiGroups: ["coordination.k8s.io"]
     resources: ["leases"]
-    verbs: ["get", "watch"]
+    verbs: ["get", "watch", "list"]
```

Comment:
The error is about listing in the `kube-node-leases` namespace:

```
karpenter-7d9786b74-4m227 controller 2023-09-06T07:21:09.828Z	ERROR	controller	pkg/mod/k8s.io/client-go@v0.26.6/tools/cache/reflector.go:169: Failed to watch *v1.Lease: failed to list *v1.Lease: leases.coordination.k8s.io is forbidden: User "system:serviceaccount:kube-system:karpenter" cannot list resource "leases" in API group "coordination.k8s.io" in the namespace "kube-node-lease": access undecided system:serviceaccount:kube-system:karpenter/[system:serviceaccounts system:serviceaccounts:kube-system system:authenticated]
```

So this should be on the clusterrole, not on the role local to kube-system afaict.

Reply:
Rather it should be a dedicated role in kube-node-leases namespace as done in the upstream PR: https://github.com/aws/karpenter/pull/4453/files

Reply:
Ah I see. I'll then match with the upstream template to create an additional role for the namespace. Thanks.